docs: [IAC-2044]: Rearrange IACM OPA usage doc (#6962)
harness/developer-hub: docs/infra-as-code-management/workspaces/project-setup/opa-workspace.md

:::

Using OPA in your Infrastructure as Code Management (IaCM) workspace gives you control over what infrastructure changes are made, for example, to add restrictions such as ensuring that a server instance is under a specified size, or that the estimated cost of a proposed change does not exceed a certain amount.

Harness IaCM allows you to use OPA on the entities that are described in the table below:

| Type/Entity | Event/Action |  Description |
| ----------- | -----------  | -------------|  
| Workspace   | On save      | This policy will be evaluated whenever there is a configuration change in a workspace (for example Terraform version, repository details, the value of a variable is updated, etc) |
| Terraform Plan   | After Terraform Plan     | This policy will be evaluated whenever a `Terraform plan` operation is performed in the IaCM stage (for example plan step, apply/destroy step, etc.). The policy will be evaluated against the Plan schema |

> | Terraform State   | After Terraform Plan | This policy will be evaluated after a `Terraform plan` operation is performed in the IaCM stage, against the state file. You can use this event to validate policy on resources, **before** applying any changes   |

Reviewer: ```suggestion
| Terraform State   | After Terraform Plan | This policy will be evaluated after a `Terraform plan` operation is performed in the IaCM stage, against the state file. You can use this event to validate policy on resources, **before** applying any changes.   |
```

> | Terraform State   | After Terraform Apply  | This policy will be evaluated after a `Terraform apply` operation is performed in the IaCM stage, against the state file. You can use this event to validate policy on resources, **after** applying any changes |

Reviewer: ```suggestion
| Terraform State   | After Terraform Apply  | This policy will be evaluated after a `Terraform apply` operation is performed in the IaCM stage, against the state file. You can use this event to validate policy on resources, **after** applying any changes. |
```

The following attributes are available to access and write policies against

**Attributes:**

> - **account**: the account identifier of the workspace.
> - **org**: the org identifier of workspace.
> - **project**: the project identifier of the workspace.
> - **created**: created timestamp.
> - **updated**: updated timestamp.
> - **description**: workspace description.
> - **name**: name of the workspace.
> - **provider_connector**: the connector for the infrastructure provider. The exact attributes within this object change depending on the type of connector.
> - **provisioner**: the provisioner (terraform).
> - **provisioner_version**: the provisioner version.
> - **repository**: the repository we pull the IAC from.
> - **repository_commit**: the commit or tag we pull from the IAC code from. Should be null if repository_branch is specified.
> - **repository_branch**: the branch we pull from the IAC code from. Should be null if repository_commit is specified.
> - **repository_connector**: the connector used to pull the IAC. The exact attributes within this object change depending on the type of connector.
> - **status**: the workspace status.
> - **environment_variables**: a map of the environment variables.
> - **terraform_variables**: a map of the terraform variables.

Reviewer: ```suggestion
- **account**: The account identifier of the workspace.
- **org**: The org identifier of workspace.
- **project**: The project identifier of the workspace.
- **created**: Created timestamp.
- **updated**: Updated timestamp.
- **description**: Workspace description.
- **name**: Name of the workspace.
- **provider_connector**: The connector for the infrastructure provider. The exact attributes within this object change depending on the type of connector.
- **provisioner**: The provisioner (terraform).
- **provisioner_version**: The provisioner version.
- **repository**: The repository from which Harness pulls the IAC.
- **repository_commit**: The commit or tag Harness pulls from the IAC code. Should be null if repository_branch is specified.
- **repository_branch**: The branch from which Harness pulls from the IAC code. Should be null if repository_commit is specified.
- **repository_connector**: The connector used to pull the IAC. The exact attributes within this object change depending on the type of connector.
- **status**: The workspace status.
- **environment_variables**: A map of the environment variables.
- **terraform_variables**: A map of the terraform variables.
```
